### DISCUSSION TITLE
How to set the Optimization Level via premake?

### DISCUSSION THREAD

**Discussion by danalemaytelescope** [premake/premake-core#2080] [Q&A]
Greetings,

Looking to set the optimization level, in project properties > Configuration Properties > C/C++ > Optimization.
Is there currently support for this with premake?

**Comment by mendsley** [comment]
The `optimize` directive handles this: https://premake.github.io/docs/optimize

**Reply by danalemaytelescope** [reply]
Weird, I had tried that & all of it's flags and didn't see a change happen to the Optimization Level. It stayed at -O0

**Reply by samsinsane** [reply]
Hard to say what the issue is without seeing your script, but my initial assumption is that you have a `filter` active when you specify the optimisation level. Something like this:
```lua
filter { "some:condition" }
  someapi "somevalue"

optimize "On"
```
